echo-notifications/src/main/groovy/com/netflix/spinnaker/echo/slack/SlackNotificationService.groovy

```diff
@@ -51,12 +51,15 @@ class SlackNotificationService implements NotificationService {
   EchoResponse.Void handle(Notification notification) {
     def text = notificationTemplateEngine.build(notification, NotificationTemplateEngine.Type.BODY)
     notification.to.each {
+      def response
       String address = it.startsWith('#') ? it : "#${it}"
       if (sendCompactMessages) {
-        slack.sendCompactMessage(token, new CompactSlackMessage(text), address, true)
+        response = slack.sendCompactMessage(token, new CompactSlackMessage(text), address, true)
       } else {
-        slack.sendMessage(token, new SlackAttachment("Spinnaker Notification", text), address, true)
+        response = slack.sendMessage(token, new SlackAttachment("Spinnaker Notification", text), address, true)
       }
+      log.debug("Received response from Slack: {} {} for message '{}'. {}",
```

Comment:
Any reason to log this out on every request?
Probably not very useful for successful responses, especially since the description says this is to aid debugging of failures?

If you change it to only log when `response.status != 200` or `response.body ok==false` (see https://api.slack.com/methods/chat.postMessage) you can make it be `log.warn` to be more obvious

Reply:
(also see https://api.slack.com/incoming-webhooks)

Reply:
I'm okay with logging in the successful case, but only if this is changed to a `trace` level.

Reply:
This is logging the *response* from Slack, which might not give you what you expected even if the Slack API returned a 200 or otherwise "okay" response, so I wanted/needed it to tell me what was going on even for successful calls.  Changed to trace -- didn't realize "trace" was a thing, thought "debug" was the most verbose option?